public/store/LogTypeStore.ts

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { NotificationsStart } from 'opensearch-dashboards/public';
+import { LogType, LogTypeBase, LogTypeItem } from '../../types';
+import LogTypeService from '../services/LogTypeService';
+import { errorNotificationToast } from '../utils/helpers';
+
+export class LogTypeStore {
+  constructor(private service: LogTypeService, private notifications: NotificationsStart) {}
+
+  public async getLogType(id: string): Promise<LogTypeItem | undefined> {
+    const logTypesRes = await this.service.searchLogTypes(id);
+    if (logTypesRes.ok) {
+      const logTypes: LogType[] = logTypesRes.response.hits.hits.map((hit) => {
+        return {
+          id: hit._id,
+          ...hit._source,
+        };
+      });
+
+      return { ...logTypes[0], detectionRules: 10 };
```

Comment:
Why are we hard coding `detectionRules: 10` to 10?

Reply:
Also if it needs to be hard coded, should this be a constant?

Reply:
Changed logic to fetch the count